src/utils/webrtc/MediaDevicesManager.js

```diff
@@ -258,25 +263,43 @@ MediaDevicesManager.prototype = {
 		}
 	},
 
-	updatePreferences() {
-		const { newAudioInputList, newVideoInputList } = updateMediaDevicesPreferences(
-			this.attributes,
-			this._preferenceAudioInputList,
-			this._preferenceVideoInputList,
-		)
+	updatePreferences(kind) {
+		if (kind === 'audioinput') {
+			const newAudioInputList = promoteMediaDevice({
+				kind,
+				devices: this.attributes.devices,
+				inputList: this._preferenceAudioInputList,
+				inputId: this.attributes.audioInputId
+			})
 
-		if (newAudioInputList) {
-			this._preferenceAudioInputList = newAudioInputList
-			BrowserStorage.setItem('audioInputPreferences', JSON.stringify(newAudioInputList))
-		}
-		if (newVideoInputList) {
-			this._preferenceVideoInputList = newVideoInputList
-			BrowserStorage.setItem('videoInputPreferences', JSON.stringify(newVideoInputList))
-		}
+			if (newAudioInputList) {
+				this._preferenceAudioInputList = newAudioInputList
+				BrowserStorage.setItem('audioInputPreferences', JSON.stringify(newAudioInputList))
+			}
+			if (!BrowserStorage.getItem('audioInputDevicePreferred')) {
+				BrowserStorage.setItem('audioInputDevicePreferred', true)
+			}
+		} else if (kind === 'videoinput') {
+			const newVideoInputList = promoteMediaDevice({
+				kind,
+				devices: this.attributes.devices,
+				inputList: this._preferenceVideoInputList,
+				inputId: this.attributes.videoInputId
+			})
 
-		const devicesPreferred = BrowserStorage.getItem('devicesPreferred')
-		if (!devicesPreferred) {
-			BrowserStorage.setItem('devicesPreferred', true)
+			if (newVideoInputList) {
+				this._preferenceVideoInputList = newVideoInputList
+				BrowserStorage.setItem('videoInputPreferences', JSON.stringify(newVideoInputList))
+			}
+			if (!BrowserStorage.getItem('videoInputDevicePreferred')) {
+				BrowserStorage.setItem('videoInputDevicePreferred', true)
+			}
+
+			// FIXME: this is a workaround to remove the old key from the browser storage
+			// To be removed in the future
+			if (BrowserStorage.getItem('devicesPreferred')) {
+				BrowserStorage.removeItem('devicesPreferred')
+			}
```

Comment:
Let's move it to the `MediaSettings` mounted hook, where it initially was introduced
That way we only call it once per page reload, not each preference update